tests/unit/modules/test_mac_service.py

```diff
@@ -206,3 +248,165 @@ def test_service_keep_alive_wrong_setting(self):
 
         with patch.object(mac_service, "show", MagicMock(return_value=info)):
             assert mac_service._always_running_service(srv_name) is False
+
+    def test_service_restart_already_loaded(self):
+        mock_cmd = MagicMock(return_value=True)
+        salt_dict = {
+            "service.loaded": mock_cmd,
+            "service.stop": mock_cmd,
+            "service.start": mock_cmd,
+        }
+        with patch.dict(mac_service.__salt__, salt_dict):
+            assert mac_service.restart("com.salt") is True
+
+    def test_service_restart_not_loaded(self):
+        salt_dict = {
+            "service.loaded": MagicMock(return_value=False),
+            "service.start": MagicMock(return_value=True),
+        }
+        with patch.dict(mac_service.__salt__, salt_dict):
+            assert mac_service.restart("com.salt") is True
+
+    def test_service_restart_failed_stop(self):
+        salt_dict = {
+            "service.loaded": MagicMock(return_value=True),
+            "service.stop": MagicMock(side_effect=CommandExecutionError),
+        }
+        with patch.dict(mac_service.__salt__, salt_dict):
+            with pytest.raises(CommandExecutionError):
+                assert mac_service.restart("com.salt")
+
+    def test_service_restart_failed_start(self):
+        salt_dict = {
+            "service.loaded": MagicMock(return_value=False),
+            "service.start": MagicMock(side_effect=CommandExecutionError),
+        }
+        with patch.dict(mac_service.__salt__, salt_dict):
+            with pytest.raises(CommandExecutionError):
+                assert mac_service.restart("com.salt")
+
+    def test_service_status_no_service(self):
+        """
+        Test service status with no service found
+        """
+        with patch.object(
+            mac_service, "_get_service", MagicMock(side_effect=CommandExecutionError)
+        ):
+            assert mac_service.status("com.salt") is False
+
+    @patch.object(mac_service, "_launch_agent", lambda _: False)
+    @patch.object(mac_service, "_get_service", lambda _: {"": ""})
+    @patch.object(mac_service, "_always_running_service", lambda _: True)
+    def test_service_status_on_daemon_with_pid(self):
+        """
+        Test service status on dameon with PID.
+        """
+        mock_servie_list = '{\n\t"LimitLoadToSessionType" = "System";\n\t"Label" = "com.salt";\n\t"OnDemand" = false;\n\t"LastExitStatus" = 0;\n\t"PID" = 218;\n\t"Program" = "/opt/salt";\n\t\t"--disable-keepalive";\n\t);\n};'
```

Comment:
Do you really want "mock_servie_list" here?

Reply:
handled